packages/runtime/src/next/api-handler/handlers/webhook/site-published.ts

```diff
@@ -1,9 +1,23 @@
 import { revalidateTag } from 'next/cache'
-import { SitePublishedWebhookPayload, WebhookHandlerResult } from './types'
+import { OnPublish, SitePublishedWebhookPayload, WebhookHandlerResult } from './types'
 
-export function handleSitePublished(_payload: SitePublishedWebhookPayload): WebhookHandlerResult {
+type SitePublishedParams = {
+  onPublish?: OnPublish
+}
+
+export async function handleSitePublished(
+  _payload: SitePublishedWebhookPayload,
+  { onPublish }: SitePublishedParams
+): Promise<WebhookHandlerResult> {
   revalidateTag(MAKESWIFT_CACHE_TAG)
 
+  try {
+    await onPublish?.()
+  } catch (error) {
+    // log and ignore any error in user-provided onPublish
+    console.error(error)
```

Comment:
I'd add context to the error log here, e.g. `console.error("Unhandled exception in the 'onPublish' callback:", error)`